### PR TITLE
Username added to the account settings

### PR DESCRIPTION
**Pull request by peratik**
## Changes

- Username field added to the account settings.

## Notion's card

https://www.notion.so/santiment/Add-Username-in-Account-settings-5ff0a684d7a045ca8370b4696288fe72

## Checklist

- [x] I've performed a self-review, followed all rules from [Frontend style guide](https://www.notion.so/santiment/Front-end-style-guide-81750096b38c4bea9a29b14fd4ab8667)
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've updated the [documentation](https://github.com/santiment/academy), if necessary (Keyboard shortcuts, Account settings)
- [x] I've checked night mode, mobile & tablet screens (if have changes in UI)
- [x] I've added tests (if necessary)

## Screenshots or GIFs
image
image

### CODE REVIEW

File: src/pages/Account/AccountPage.module.scss [hunk 1]
```diff
@@ -164,6 +164,20 @@
   }
 }
 
+.prefix {
+  position: absolute;
+  color: var(--jungle-green);
+  font-size: 14px;
+  line-height: 20px;
```
Comment (DmitriVanGuard) [inline src/pages/Account/AccountPage.module.scss:171]: `@include text('body-3');` can be reused here

File: src/pages/Account/AccountPage.module.scss [hunk 2]
```diff
@@ -164,6 +164,20 @@
...
+  top: 50%;
+  transform: translateY(-50%);
+  left: 25px;
+
+  @include responsive('phone', 'phone-xs') {
```
Comment (DmitriVanGuard) [inline src/pages/Account/AccountPage.module.scss:176]: Better to use the new `.phone, .phone-xs` classes

File: src/pages/Account/EditableInputSetting.js [hunk 3]
```diff
@@ -89,15 +90,19 @@ class EditableInputSetting extends PureComponent {
                 )}
                 accent='waterloo'
               >
-                {defaultValue || `Please add your ${label.toLowerCase()}`}
+                {`${!!prefix ? prefix : ''}${defaultValue}` || `Please add your ${label.toLowerCase()}`}
```
Comment (DmitriVanGuard) [inline src/pages/Account/EditableInputSetting.js:93]: `prefix || ''`

File: src/pages/Account/name/NameSetting.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+const CHANGE_NAME_MUTATION = gql`
+  mutation changeName($value: String!) {
+    changeName(name: $value) {
+      name
+    }
+  }
+`
+
+const validateName = name => {
```
Comment (DmitriVanGuard) [inline src/pages/Account/name/NameSetting.js:17]: `[D-0-1]` For function better to use the `function` declaration. We are using `const` for components, one line functions and variables

File: src/pages/Account/name/NameSetting.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  if (name && name.length < 3) {
+    return 'Name should be at least 3 characters long'
+  }
```
Comment (DmitriVanGuard) [inline src/pages/Account/name/NameSetting.js:20]: `[D-0-2]` If the `name` is empty string/falsy, function will return `undefined`. Is it desired behaviour?

File: src/pages/Account/userName/UsernameSetting.js [hunk 6]
```diff
@@ -17,19 +17,28 @@ const CHANGE_USERNAME_MUTATION = gql`
 `
 
 const validateUsername = username => {
-  if (!username || username.length < 3) {
+  if (username && username.length < 3) {
     return 'Username should be at least 3 characters long'
   }
+  if (username && username[0] === "@") {
+    return "@ is not allowed for the first character"
+  }
```
Comment (DmitriVanGuard) [inline src/pages/Account/userName/UsernameSetting.js:25]: Same as the `[D-0-2]`

File: src/pages/Account/userName/UsernameSetting.js [hunk 7]
```diff
@@ -17,19 +17,28 @@ const CHANGE_USERNAME_MUTATION = gql`
...
 }
 
-const UsernameSetting = ({ dispatchNewUsername, username, changeUsername }) => {
+const UsernameSetting = ({ dispatchNewUsername, username, name, changeUsername }) => {
+  let _username = username;
+  if (!username && name) {
+    _username = name.toLowerCase().replace(/ /g,"_");
+  }
+
```
Comment (DmitriVanGuard) [inline src/pages/Account/userName/UsernameSetting.js:33]:
```js
const normalizedUsername = username || name.toLowerCase().replace(/ /g,"_");
```

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Account/name/NameSetting.js (hunk 5 above)
Comment (peratik) [reply src/pages/Account/name/NameSetting.js:20]: Yes names and usernames are nullable.

File: src/pages/Account/userName/UsernameSetting.js (hunk 7 above)
Comment (peratik) [reply src/pages/Account/userName/UsernameSetting.js:33]: There is another case: `!username && !name` so we can't use `||` here.

File: src/pages/Account/AccountPage.module.scss (hunk 2 above)
Comment (peratik) [reply src/pages/Account/AccountPage.module.scss:176]: Can you please describe more? Hoe can we use them?

File: src/pages/Account/userName/UsernameSetting.js (hunk 6 above)
Comment (peratik) [reply src/pages/Account/userName/UsernameSetting.js:25]: Yes names and usernames are nullable.

File: src/pages/Account/AccountPage.module.scss (hunk 2 above)
Comment (DmitriVanGuard) [reply src/pages/Account/AccountPage.module.scss:176]: @peratik 
```scss
:global(.phone) &, :global(.phone-xs) & {
    left: 10px;
}
```

File: src/pages/Account/userName/UsernameSetting.js (hunk 7 above)
Comment (DmitriVanGuard) [reply src/pages/Account/userName/UsernameSetting.js]: `... = username || (name && name.toLower... )`

File: src/stores/user/index.js [hunk 8]
```diff
@@ -9,6 +9,7 @@ export const USER_QUERY = gql`
     currentUser {
       id
       username
+      name
```
Comment (Partysun) [inline src/stores/user/index.js:12]: Can we add comments here, what is the difference between name and username @peratik

Comment (peratik) [reply src/stores/user/index.js]: Added.

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**